update.py

```diff
@@ -219,6 +228,58 @@ def copy_build(site):
         shutil.rmtree(olddir)
 
 
+def copy_and_keep_build(site):
+    """
+    Copies each site into the target location and keep last "n" builds as backups
+    """
+    global error_count
+    for wiki in ALL_WIKIS:
+        if site=='common':
+            continue
+        if site is not None and site != wiki:
+            continue
+        debug('coping: %s' % wiki)
+        targetdir = os.path.join(args.destdir, wiki)
+
+        if os.path.exists(targetdir):
+            olddir = os.path.join(args.backupdestdir, str(building_time), str(wiki) )
+            debug('Checking %s' % olddir )
+            distutils.dir_util.mkpath(olddir)
+            debug('Moving %s into %s' % (targetdir,olddir))
+            shutil.move(targetdir, olddir)
+
+        sourcedir = os.path.join(os.path.abspath(os.getcwd()), str(wiki), 'build', 'html')
+        html_moved_dir = os.path.join(args.destdir, 'html')
+        try:
+            #subprocess.check_call(['mv', sourcedir, html_moved_dir])
+            shutil.move(sourcedir, html_moved_dir)
+            # Rename move! (single move to html/* failed)
+            #subprocess.check_call(['mv', html_moved_dir ,targetdir])
+            shutil.move(html_moved_dir, targetdir)
+            debug("Moved to %s" % targetdir)
+        except:
+            error("FAIL moving output to %s" % targetdir)
+        finally:
+            debug("Creating a backup in %s" % olddir)
+            # subprocess.check_call(['cp', '-r', targetdir ,olddir])
+            distutils.dir_util.copy_tree(targetdir, olddir, preserve_symlinks=0)
+
+
+def delete_old_wiki_backups(folder, n_to_keep):
+    try:
+        debug('Checking number of number of backups in folder %s' % folder)
+        backup_folders = glob.glob(folder  + "/*/")
```

Comment:
Could we tighten the glob here somehow?  As it stands someone can pass in a bad archive folder and nuke stuff they didn't want to nuke.  I'm usually an "enough rope" sort of person, but this sort of construct makes me a bit wary.

Something like `os.path.join(folder, "*-*-*-*-*/")` might work, for example.

Reply:
I do agree because I don't trust users, especially when I'm one of them :-)

I forced a nomenclature in the folders and when it was time to delete them.